src/dotnet-interactive/CommandLine/CommandLineParser.cs

```diff
@@ -304,15 +328,36 @@ Command StdIO()
                                 }, kernel =>
                                 {
                                     StdIOCommand.CreateServer(kernel, console);
+
+                                    cancellationToken.Register(async () =>
+                                    {
+                                        await kernel.SendAsync(new Quit());
+                                    });
                                 });
 
                             return startHttp(startupOptions, console, startServer, context);
                         }
 
-                        return startStdIO(
-                            startupOptions,
-                            CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(), startupOptions),
-                            console);
+                        {
+                            var kernel = CreateKernel(options.DefaultKernel, new BrowserFrontendEnvironment(),
+                                startupOptions);
+                            
+                            Quit.DefaultOnQuit = () =>
```

Comment:
Any reason not to hardcode the default quit behavior?

Reply:
that is the only time i am overriding it